tests/test_tables.py

```diff
@@ -251,3 +251,18 @@ def test_rename_columns_inplace() -> None:
     t.rename(columns={"gdp": "new_gdp"}, inplace=True)
     assert t.new_gdp.metadata.title == "GDP"
     assert t.columns == ["new_gdp"]
+
+
+def test_copy() -> None:
+    t: Table = Table({"gdp": [100, 102, 104], "country": ["AU", "SE", "CH"]}).set_index("country")  # type: ignore
+    t.metadata.title = "GDP table"
+    t.gdp.metadata.title = "GDP"
+    t2 = t.copy()
+
+    t2.metadata.title = "GDP table copy"
+    t2.gdp.metadata.title = "GDP copy"
+
+    assert t.gdp.metadata.title == "GDP"
```

Comment:
Maybe it would be good to also do a similar test but with a mutable object, for example, `primary_key`.

Reply:
Good idea, added